Products/ZenCollector/scheduler.py

```diff
@@ -505,12 +508,23 @@ def getTasksForConfig(self, configId):
 
     def getNextExpectedRun(self, taskName):
         """
-        Get the next expected execution time for given task
+        Get the next expected execution time for given task.
         """
         loopingCall = self._loopingCalls.get(taskName, None)
         if loopingCall:
             return loopingCall._expectNextCallAt
 
+    def setNextExpectedRun(self, taskName, taskInterval):
+        """
+        Set the next expected execution time for given task.
+        """
+        loopingCall = self._loopingCalls[taskName]
+        loopingCall._expectNextCallAt = time.time() + taskInterval
```

Comment:
It shouldn't be the case according to the execution order, however, I've updated the branch, please recheck.